lib/iris/tests/unit/constraints/test_Constraint_equality.py

```diff
@@ -0,0 +1,266 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""Unit tests for equality testing of different constraint types."""
+
+from __future__ import absolute_import, division, print_function
+
+from six.moves import filter, input, map, range, zip  # noqa
+
+from iris._constraints import AttributeConstraint, Constraint, NameConstraint
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+
+class Test_Constraint__hash__(tests.IrisTest):
+    def test_empty(self):
+        c1 = Constraint()
+        c2 = Constraint()
+        self.assertEqual(hash(c1), hash(c1))
+        self.assertNotEqual(hash(c1), hash(c2))
+
+
+class Test_Constraint__eq__(tests.IrisTest):
+    def test_empty_same(self):
+        c1 = Constraint()
+        c2 = Constraint()
+        self.assertEqual(c1, c2)
+        self.assertIsNot(c1, c2)
+
+    def test_emptyname_same(self):
+        c1 = Constraint("")
+        c2 = Constraint("")
+        self.assertEqual(c1, c2)
+
+    def test_empty_emptyname_differ(self):
+        c1 = Constraint()
+        c2 = Constraint("")
+        self.assertNotEqual(c1, c2)
+
+    def test_names_same(self):
+        c1 = Constraint("a")
+        c2 = Constraint("a")
+        self.assertEqual(c1, c2)
+
+    def test_names_differ(self):
+        c1 = Constraint("a")
+        c2 = Constraint("b")
+        self.assertNotEqual(c1, c2)
+
+    def test_funcs_same(self):
+        # *Same* functions match
+        def func(cube):
+            return False
+
+        c1 = Constraint(cube_func=func)
+        c2 = Constraint(cube_func=func)
+        self.assertEqual(c1, c2)
+
+    def test_funcs_differ(self):
+        # Identical but different funcs do not match.
+        c1 = Constraint(cube_func=lambda c: False)
+        c2 = Constraint(cube_func=lambda c: False)
+        self.assertNotEqual(c1, c2)
+
+    def test_coord_names_same(self):
+        c1 = Constraint(coord=3)
```

Comment:
I think it would be clearer to replace instances the coord name "coord" with something like "latitude". This makes it clear that the keyword represents the name of the coord and is not some special keyword like `coord_values`

Reply:
:+1: